lewis/core/devices.py

```diff
@@ -111,21 +111,25 @@ class DeviceBuilder(object):
     def __init__(self, module):
         self._module = module
 
-        self._device_types = list(get_members(self._module, is_device).values())
-
-        if not self._device_types:
-            # syslog.warning('No devices discovered in module {}.'.format(module.__name__))
-            pass
-        else:
-            pass
-            # syslog.info('Discovered devices in module {}: {}'.format(
-            #    module.__name__, ','.join([x.__name__ for x in self._device_types])))
-
         submodules = get_submodules(self._module)
 
+        # Devices can be in __init__.py, device.py, or any devices/*.py
+        self._device_types = list(get_members(self._module, is_device).values())
+        self._device_types += list(get_members(submodules.get('device'), is_device).values())
```

Comment:
That is actually true, it's slightly different. But yes, there must be a way to make it consistent (probably other ways than how it's implemented in the interface discovery).